dvc/repo/metrics/diff.py

```diff
@@ -19,8 +17,12 @@ def diff(repo, *args, a_rev=None, b_rev=None, **kwargs):
 
     with_unchanged = kwargs.pop("all", False)
 
-    old = _get_metrics(repo, *args, **kwargs, rev=(a_rev or "HEAD"))
-    new = _get_metrics(repo, *args, **kwargs, rev=b_rev)
+    a_rev = a_rev or "HEAD"
+    b_rev = b_rev or "workspace"
```

Comment:
Note that the changes here are not equivalent, as the previous one would default to the `repo.tree`'s state whereas here it's the workspace.

Reply:
If we want to make it equivalent, we will likely need a knob in the `brancher` to not yield the "workspace".

Reply:
Considering that help message (for `metrics diff`) mentions that `b_rev` defaults to `workspace`, I would say its actually bug fixed. Also, `plots diff` defaults to `workspace`.